### PR TITLE
[full-ci] Enhancement: Show text file icon for empty text files

### PR DESCRIPTION
**Pull request by lookacat**
## Description
When having an almost empty txt file the thumbnail shows just a white square which looks weird.
We now replaced the thumbnail with the txt file icon for almost empty files.

## Motivation and Context
Make UI not look bugged

## Screenshots (if appropriate):
Before (all empty)
empty

After (all empty have txt file icon, the one with content displays a thumbnail)
after

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingIntGroupsToRoot</strong> https://drone.owncloud.com/owncloud/web/30292/26/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUISharingInternalGroupsToRoot-shareWithGroups_feature-L203.png
  


</p></details>

File: packages/design-system/src/components/OcResource/OcResource.vue [hunk 1]
```diff
@@ -165,9 +165,14 @@ export default {
       }
     },
 
+    isTxtFileAlmostEmpty() {
```
Comment (kulmann) [inline packages/design-system/src/components/OcResource/OcResource.vue:168]: IMO this logic doesn't belong into the `OcResource` component - it already has a prop whether or not the thumbnail should be displayed. We should make use of the prop from wherever we embed the `OcResource` component (currently in `ResourceTable.vue` and search `Preview.vue`). Ping me if you want to work on this together.

**Review by kulmann — CHANGES_REQUESTED** [review]

File: packages/design-system/src/components/OcResource/OcResource.vue (hunk 1 above)
Comment (lookacat) [reply packages/design-system/src/components/OcResource/OcResource.vue:168]: alright

File: packages/web-pkg/src/helpers/preview/textFilePreview.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { Resource } from 'web-client/src'
+
+export const isResourceTxtFileAlmostEmpty = (resource: Resource): boolean => {
```
Comment (kulmann) [inline packages/web-pkg/src/helpers/preview/textFilePreview.ts:3]: my last two nitpicks:
- this is a resource helper that has nothing to do with previews anymore. The file name and location should reflect that accordingly. This only states if a text file is almost empty. Nothing more, nothing less.
- please add a unit test for the helper

**Review by kulmann — CHANGES_REQUESTED** [review]

File: packages/web-pkg/src/helpers/preview/textFilePreview.ts (hunk 2 above)
Comment (lookacat) [reply packages/web-pkg/src/helpers/preview/textFilePreview.ts:3]: yup

**Review by kulmann — APPROVED** [review]

**Pull request merged**